mudlet-mapper.xml

```diff
@@ -9153,6 +9175,197 @@ registerAnonymousEventHandler(&quot;mmapper arrived&quot;, &quot;mmp.WearShackle
 registerAnonymousEventHandler(&quot;started speedwalk&quot;, &quot;mmp.UnwearShackle&quot;)</script>
                         <eventHandlerList/>
                     </Script>
+                    <Script isActive="yes" isFolder="no">
+                        <name>mmapper_achaea_orb_of_confinement</name>
+                        <packageName></packageName>
+                        <script>local cityIDs =
+  {
+    [49] =
+      function()
+        if mmp.settings.orbashtan then
+          mmp.echo(&quot;Orb of Confinement set as &lt;green&gt;ACTIVE&lt;reset&gt; for Ashtan&quot;)
+        else
+          mmp.echo(&quot;Orb of Confinement set as &lt;red&gt;INACTIVE&lt;reset&gt; for Ashtan&quot;)
+        end
+      end,
+    [57] =
+      function()
+        if mmp.settings.orbcyrene then
+          mmp.echo(&quot;Orb of Confinement set as &lt;green&gt;ACTIVE&lt;reset&gt; for Cyrene&quot;)
+        else
+          mmp.echo(&quot;Orb of Confinement set as &lt;red&gt;INACTIVE&lt;reset&gt; for Cyrene&quot;)
+        end
+      end,
+    [51] =
+      function()
+        if mmp.settings.orbeleusis then
+          mmp.echo(&quot;Orb of Confinement set as &lt;green&gt;ACTIVE&lt;reset&gt; for Eleusis&quot;)
+        else
+          mmp.echo(&quot;Orb of Confinement set as &lt;red&gt;INACTIVE&lt;reset&gt; for Eleusis&quot;)
+        end
+      end,
+    [55] =
+      function()
+        if mmp.settings.orbhashan then
+          mmp.echo(&quot;Orb of Confinement set as &lt;green&gt;ACTIVE&lt;reset&gt; for Hashan&quot;)
+        else
+          mmp.echo(&quot;Orb of Confinement set as &lt;red&gt;INACTIVE&lt;reset&gt; for Hashan&quot;)
+        end
+      end,
+    [44] =
+      function()
+        if mmp.settings.orbmhaldor then
+          mmp.echo(&quot;Orb of Confinement set as &lt;green&gt;ACTIVE&lt;reset&gt; for Mhaldor&quot;)
+        else
+          mmp.echo(&quot;Orb of Confinement set as &lt;red&gt;INACTIVE&lt;reset&gt; for Mhaldor&quot;)
+        end
+      end,
+    [368] =
+      function()
+        if mmp.settings.orbtargossas then
+          mmp.echo(&quot;Orb of Confinement set as &lt;green&gt;ACTIVE&lt;reset&gt; for Targossas&quot;)
+        else
+          mmp.echo(&quot;Orb of Confinement set as &lt;red&gt;INACTIVE&lt;reset&gt; for Targossas&quot;)
+        end
+      end,
+  }
+
+function mmp.setOrb(areaID)
```

Comment:
I'd probably rewrite this function like this:
```lua
function mmp.setOrb(city)
  mmp.clearpathcache()
  if mmp.settings["orb"..city:lower()] then
    mmp.echo("Orb of Confinement set as <green>ACTIVE<reset> for " .. city:title())
  else
    mmp.echo("Orb of Confinement set as <red>INACTIVE<reset> for " .. city:title())
  end
end
```

and have the lambda in the `createOption` call be
```lua
function() mmp.setOrb("targossas") end
```

I think that makes the call cleaner and gets rid of the `cityIDs` table which includes a lot of duplicated code.

Reply:
Made the changes as requested